docs/sdk/src/reference_docs/frame_origin.rs

```diff
@@ -1,14 +1,260 @@
 //! # FRAME Origin
 //!
-//! Notes:
-//!
-//! - Def talk about account abstraction and how it is a solved issue in frame. See Gav's talk in
-//!   Protocol Berg 2023
-//! - system's raw origin, how it is amalgamated with other origins into one type
-//! [`frame_composite_enums`]
-//! - signed origin
-//! - unsigned origin, link to [`fee_less_runtime`]
-//! - Root origin, how no one can obtain it.
-//! - Abstract origin: how FRAME allows you to express "origin is 2/3 of the this body or 1/2 of
-//!   that body or half of the token holders".
-//! - `type CustomOrigin: EnsureOrigin<_>` in pallets.
+//! let's start by eliminating a wrong assumption that is a common mistake:
+//!
+//! **ORIGIN IS NOT AN ACCOUNT ID**.
+//!
+//! FRAME's origin abstractions allow you to convey meanings far beyond just an account-id being the
+//! caller of an extrinsic. Nonetheless, an account-id having signed an extrinsic is one of the
+//! meanings that an origin can convey. This is the commonly used [`frame_system::ensure_signed`],
+//! where the return value happens to be an account-id.
+//!
+//! Instead, let's establish the following as the correct definition of an origin:
+//!
+//! > The origin type represents the privilege level of the caller of an extrinsic.
+//!
+//! That is, an extrinsic, through checking the origin, can *express what privilege level it wishes
+//! to impose on the caller of the extrinsic*. One of those checks can be as simple as "*any account
+//! that has signed a statement can pass*".
+//!
+//! But the origin system can also express more abstract and complicated privilege levels. For
+//! example:
+//!
+//! * If the majority of token holders agreed upon this. This is more or less what the
+//!   [`pallet_democracy`] does under the hood ([reference](https://github.com/paritytech/polkadot-sdk/blob/edd95b3749754d2ed0c5738588e872c87be91624/substrate/frame/democracy/src/lib.rs#L1603-L1633)).
```

Comment:
Is this an example of using the Origin abstraction? Seems just like regular logic

Reply:
It is meant to showcase that pallets can dispatch internally and set any origin. And, that an origin like root can only be acquired in this way. 

I will re-read it one more time.